src/java/org/apache/cassandra/index/sai/plan/Expression.java

```diff
@@ -333,22 +335,89 @@ public boolean isSatisfiedBy(ByteBuffer columnValue)
         return true;
     }
 
-    public ByteBuffer getLowerBound()
+    /**
+     * Returns the lower bound of the expression as a ByteComparable with an encoding based on the version, the
+     * validator, and whether the expression is in memory or on disk.
+     * @param version the version of the index
+     * @param inMemory whether the expression is in memory or on disk
+     * @return
+     */
+    public ByteComparable getEncodedLowerBoundByteComparable(Version version, boolean inMemory)
     {
-        return getBound(lower, true);
+        // Note: this value was encoded using the TypeUtil.encode method, but it wasn't
+        var bound = getPartiallyEncodedLowerBound(version);
+        if (bound == null)
+            return null;
+        var terminator = lower.inclusive ? ByteSource.TERMINATOR : ByteSource.GT_NEXT_COMPONENT;
```

Comment:
Use `ByteSource.LT_NEXT_COMPONENT` for inclusive left to make sure the bound is not a prefix of some key (which may make reverse iteration more difficult) and align with the way we define boundaries for clustering keys.

Reply:
Thanks for catching that nuance.